src/openvic-simulation/country/CountryInstance.cpp

```diff
@@ -220,27 +243,37 @@ bool CountryInstance::set_upper_house(Ideology const* ideology, fixed_point_t po
 	}
 }
 
-bool CountryInstance::add_reform(Reform const* new_reform) {
-	if (std::find(reforms.begin(), reforms.end(), new_reform) != reforms.end()) {
-		Logger::warning(
-			"Attempted to add reform \"", new_reform, "\" to country ", get_identifier(), ": already present!"
-		);
-		return false;
+bool CountryInstance::set_ruling_party(CountryParty const& new_ruling_party) {
+	if (ruling_party != &new_ruling_party) {
+		ruling_party = &new_ruling_party;
+
+		return update_rule_set();
+	} else {
+		return true;
 	}
-	reforms.push_back(new_reform);
-	return true;
 }
 
-bool CountryInstance::remove_reform(Reform const* reform_to_remove) {
-	auto existing_entry = std::find(reforms.begin(), reforms.end(), reform_to_remove);
-	if (existing_entry == reforms.end()) {
-		Logger::warning(
-			"Attempted to remove reform \"", reform_to_remove, "\" from country ", get_identifier(), ": not present!"
-		);
-		return false;
+bool CountryInstance::add_reform(Reform const& new_reform) {
+	ReformGroup const& reform_group = new_reform.get_reform_group();
+	decltype(reforms)::value_ref_t reform = reforms[reform_group];
+
+	if (reform != &new_reform) {
+		if (reform_group.is_administrative()) {
+			if (reform != nullptr) {
+				total_administrative_multiplier -= reform->get_administrative_multiplier();
```

Comment:
We could recalculate it every day, but this approach gives the same result with fewer calculations needed.
And the +/- won't have any problems from accuracy, fixed points can have error when they're first parsed but after that they're just integers when being added. Whenever we subtract a value it'll be one that was added to the total earlier, the exact same value from the same variable in the same reform. This will always produce the same value as adding them all from scratch, if there's any error from fixed point parsing inaccuracy then it'll show up in the same way with either approach.